File: tests/graph/eval/fixtures/spring_boot_sample/src/main/java/com/example/blog/service/UserService.java
package com.example.blog.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.blog.entity.User;
import com.example.blog.repository.UserRepository;

@Service
public class UserService {
    private final UserRepository userRepository;

    @Autowired
    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User register(String email, String name) {
        if (userRepository.findByEmail(email).isPresent()) {
            throw new IllegalStateException("email already in use");
        }
        User user = new User();
        user.setEmail(email);
        user.setName(name);
        return userRepository.save(user);
    }

    public User get(Long id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("user not found"));
    }
}

File: tests/graph/eval/fixtures/kotlin_spring_sample/src/main/kotlin/com/example/blog/service/UserService.kt
package com.example.blog.service

import com.example.blog.entity.User
import com.example.blog.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(private val userRepository: UserRepository) {

    fun register(email: String, name: String): User {
        val user = User()
        user.email = email
        user.name = name
        return userRepository.save(user)
    }

    fun get(id: Long): User? = userRepository.findById(id).orElse(null)
}
